docker/maistra-builder_1.2.Dockerfile

```diff
@@ -39,11 +44,22 @@ RUN curl https://get.helm.sh/helm-${HELM_VERSION}-linux-amd64.tar.gz | tar -xz l
 # Hugo
 RUN curl -L https://github.com/gohugoio/hugo/releases/download/v${HUGO_VERSION}/hugo_${HUGO_VERSION}_Linux-64bit.tar.gz | tar -xz hugo && mv hugo /usr/local/bin
 
-
 # Kubectl
 RUN curl -sfL https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl -o /usr/local/bin/kubectl && \
     chmod +x /usr/local/bin/kubectl
 
+# KinD
+RUN curl -Lo /usr/local/bin/kind https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-linux-amd64 && \
+    chmod +x /usr/local/bin/kind
+
+# gcloud
+# some upstream scripts rely on gcloud commands to retrieve credentials for
+# e.g. pushing to GCS buckets. We don't require this at this point, so we're
+# only creating a stub script to make sure scripts don't fail because gcloud
+# is missing.
+RUN echo 'echo "gcloud is not installed. Skipping"' > /usr/local/bin/gcloud && \
+    chmod +x /usr/local/bin/gcloud
+
```

Comment:
Really? Those should be optional, like here: https://github.com/istio/istio/blob/master/prow/lib.sh#L32
If they aren't please, fix that upstream :)

Reply:
They are now, but were not in 1.4: https://github.com/istio/istio/blob/cf35a864ff65ae3a6d0278fa0c5bfead0eaafffe/prow/lib.sh#L38

I guess we could fix it in our fork, too. But I don't think changing that upstream now makes a lot of sense

Reply:
We should not be doing that, that's the good thing about using branches, one can evolve independently.